src/java.base/share/classes/java/lang/reflect/Method.java

```diff
@@ -93,7 +96,19 @@ public final class Method extends Executable {
     private Method root;
     private transient volatile MethodRepository genericInfo;
     private @Stable MethodAccessor methodAccessor;
+
+    private static final ToIntFunction<Method> HASH_UPDATER =
+            StableFieldUpdater.ofIntRaw(Method.class, Unsafe.getUnsafe().objectFieldOffset(Method.class, "hash"), new ToIntFunction<Method>() {
```

Comment:
If we are using unsafe to get field offsets, why don't we just accept arguments to locate a MethodHandle (like those arguments to MhUtils)?

Reply:
Do you mean for interacting with the field or computing the hash?

Reply:
Calling the method that computes hash.

Reply:
This would require another implementation to be written. Maybe we can add that later?

Reply:
I feel like declaring a function is declaring redundant classes when we can just reuse a DirectMethodHandle pointing to an actual method. Also the zero alternative should be handled by the provided computing function/method instead of handled by this utility.

Reply:
There is an MH variant now. The use of the zero alternative is optional, as if you put in `0`, it will have no effect.